src/mesh/gmsh_reader.cpp

```diff
@@ -438,10 +501,117 @@ unsigned int ij_to_num(const unsigned int i,
     return i + j*n_per_line;
 }
 
+/**
+ * Solely used in the 3D case. Helps finding the face_nodes during recursive call.
+ * @param degree
+ * @return
+ */
+std::vector<unsigned int>
+face_node_finder(const unsigned int degree, const bool mesh_reader_verbose_output)
+{
+
+    /**
+    * For Debug output
+    */
+    const int mpi_rank = dealii::Utilities::MPI::this_mpi_process(MPI_COMM_WORLD);
+    dealii::ConditionalOStream pcout(std::cout, mpi_rank==0);
+
+    // number of support points in each direction
+    const unsigned int n = degree + 1;
+
+    const unsigned int dofs_per_cell = dealii::Utilities::fixed_power<2>(n);
+
+    std::vector<unsigned int> h2l_2D(dofs_per_cell);
+    if (degree == 0) {
+        h2l_2D[0] = 0;
+        return h2l_2D;
+    }
+
+    // polynomial degree
+    const unsigned int dofs_per_line = degree - 1;
```

Comment:
@kit71717 is this a bug?

Reply:
Fixed in the latest commit, it was not being used in this function, hence I removed it. It originated from 
the function "gmsh_hierarchic_to_lexicographic", I changed "dofs_per_line" to "inner_points_per_line" to make it less ambiguous since the original intent is to loop through inner points, i.e., 1-2-3-4, and we want to extract 2-3.